src/mds/MDSMap.cc

```diff
@@ -177,11 +177,20 @@ void MDSMap::dump(Formatter *f) const
     f->close_section();
   }
   f->close_section();
-  f->open_array_section("data_pools");
-  for (set<int64_t>::const_iterator p = data_pools.begin(); p != data_pools.end(); ++p)
-    f->dump_int("pool", *p);
-  f->close_section();
-  f->dump_int("metadata_pool", metadata_pool);
+
+  if (data_pools.empty())
+    f->dump_string("data_pools", "None");
```

Comment:
Let's not output None here: an empty list (the current behaviour) is more appropriate